### PR TITLE
UDF for ASL Recognition

### PR DESCRIPTION
**Pull request by IshSiva**
- [x] Train a model for ASL Recognition
- [x] Create a new UDF for ASL Recognition
- [x] Create a new tutorial notebook for actions

### CODE REVIEW

File: eva/udfs/asl_action_recognition.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+    @property
+    def name(self) -> str:
+        return "ASLActionRecognition"
+
+    
+
+    def download_weights(self):
+        if not os.path.exists(self.asl_weights_path):
+            torch.hub.set_dir("/nethome/gkakkar7/isiva/eva_codebase/eva/eva/udfs/")
```
Comment (jarulraj) [inline eva/udfs/asl_action_recognition.py:47]: We will need to remove such hard-coded paths

**Review by jarulraj — CHANGES_REQUESTED** [review]

File: eva/udfs/asl_action_recognition.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+import pandas as pd
+import torch
+import torchvision
+
+try:
+    from torchvision.models.video import R3D_18_Weights, r3d_18
+
+except ImportError:
+    raise ImportError(
+        f"torchvision>=0.14.0 is required to use MVITActionRecognition, found {torchvision.__version__}"
```
Comment (gaurav274) [inline eva/udfs/asl_action_recognition.py:29]: Is this comment correct?

File: eva/udfs/asl_action_recognition.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            "https://gatech.box.com/shared/static/crjhyy4nc2i5nayesfljutwc1y3bpw2q.pth"
+        )
+        self.asl_weights_path = os.getcwd() + "/asl_weights.pth"
+        self.download_weights()
+
+        self.weights = R3D_18_Weights.DEFAULT
+        self.model = r3d_18(weights=self.weights)
+        in_feats = self.model.fc.in_features
+        self.model.fc = nn.Linear(in_feats, 20)
+        self.model.load_state_dict(torch.load(self.asl_weights_path))
```
Comment (gaurav274) [inline eva/udfs/asl_action_recognition.py:65]: See this to load on cpu by default.
 https://pytorch.org/docs/stable/generated/torch.load.html

File: eva/udfs/asl_action_recognition.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+
+class ASLActionRecognition(PytorchAbstractClassifierUDF):
+    @property
+    def name(self) -> str:
+        return "ASLActionRecognition"
+
+    def download_weights(self):
+        if not os.path.exists(self.asl_weights_path):
+            torch.hub.set_dir(os.getcwd())
```
Comment (gaurav274) [inline eva/udfs/asl_action_recognition.py:46]: Is this a persistent change? We don't want to after the default torch behavior

File: eva/udfs/asl_action_recognition.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+                self.asl_weights_path,
+                hash_prefix=None,
+                progress=True,
+            )
+
+    def setup(self):
+        self.asl_weights_url = (
+            "https://gatech.box.com/shared/static/crjhyy4nc2i5nayesfljutwc1y3bpw2q.pth"
+        )
+        self.asl_weights_path = os.getcwd() + "/asl_weights.pth"
```
Comment (gaurav274) [inline eva/udfs/asl_action_recognition.py:58]: Consider using this https://github.com/georgia-tech-db/eva/blob/04040765a43bc43e699feacb461d3b063bcbfb08/eva/configuration/constants.py#L30

File: eva/udfs/asl_action_recognition.py (hunk 2 above)
Comment (IshSiva) [reply eva/udfs/asl_action_recognition.py]: updated

File: eva/udfs/asl_action_recognition.py (hunk 3 above)
Comment (IshSiva) [reply eva/udfs/asl_action_recognition.py]: updated to CPU

File: eva/udfs/asl_action_recognition.py (hunk 5 above)
Comment (IshSiva) [reply eva/udfs/asl_action_recognition.py]: updated.

**Comment by jarulraj** [comment]
@IshSiva  Can you add a few frames from the ASL video with the keyword annotation similar to other notebooks?

File: eva/udfs/asl_action_recognition.py (hunk 4 above)
Comment (gaurav274) [reply eva/udfs/asl_action_recognition.py:47]: We can remove this right?

Comment (IshSiva) [reply eva/udfs/asl_action_recognition.py]: removed this.

File: test/integration_tests/test_pytorch.py [hunk 6]
```diff
@@ -42,11 +46,14 @@ def setUpClass(cls):
         ua_detrac = f"{EVA_ROOT_DIR}/data/ua_detrac/ua_detrac.mp4"
         mnist = f"{EVA_ROOT_DIR}/data/mnist/mnist.mp4"
         actions = f"{EVA_ROOT_DIR}/data/actions/actions.mp4"
+        asl_actions = f"{EVA_ROOT_DIR}/data/actions/computer_asl.avi"
```
Comment (jarulraj) [inline test/integration_tests/test_pytorch.py:49]: Is this file uploaded? Is it redundant with "computer_asl.mp4"?

Comment (IshSiva) [reply test/integration_tests/test_pytorch.py:49]: i have updated the tests to use computer_asl.mp4. the file is also uploaded.

**Review by gaurav274 — APPROVED** [review]

**Comment by jarulraj** [comment]
Looks like the notebook is broken.

**Pull request merged**